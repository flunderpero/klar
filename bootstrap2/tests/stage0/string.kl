enum Color:
    Red
    Green
    Blue
    Orange(intensity i32)
end

struct Person:
    name str
    age i32
    favorite_color Color
end

fn person_to_str(p Person) str:
    return f"Person: name={p.name}, age={p.age}"
end

fn main():
    let name = "Mary"
    let i = 42
    let b = true
    let p = Person("Mary", 42)
    print(f"str: Hi {name}")
    print(f"i32: {i}")
    print(f"bool: {b}")
    print(f"bool: {false}")
    print(f"enum (Red): {Color.Red}")
    print(f"enum (Green): {Color.Green}")
    print(f"enum (Blue): {Color.Blue}")
    print(f"enum (Orange): {Color.Orange(42)}")
    print(f"struct members: name={p.name}, age={p.age}")
    print(f"function call: {person_to_str(p)}")
    print(f"multiple: Hi {name}, the i32 is {i}, the bool is {b}, and {p.name} is {p.age} years old.")
end
