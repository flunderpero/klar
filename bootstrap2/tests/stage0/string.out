str: Hi Mary
i32: 42
bool: true
bool: false
enum (Red): 0
enum (Green): 1
enum (Blue): 2
enum (Orange): 3
struct members: name=Mary, age=42
function call: Person: name=Mary, age=42
multiple: Hi Mary, the i32 is 42, the bool is true, and Mary is 42 years old.
